app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_ = {}
+
+    def inner(*args) -> Any:
+        key = args
+        if key in cache_:
+            print("Getting from cache")
+            return cache_[key]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_[key] = result
+            return result
```

Comment:
Think how you can call `return` only once . If `cache_[key] = result` you can `return cached[key]` and it is makes 2 identical returns in your func . Optimise it : )